docs/api_reference/src/quantities.tex

```diff
@@ -3107,40 +3101,140 @@
 
 template<typename T>
-concept @\defexposconceptnc{WeaklyRegular}@ = std::@\stdconcept{copyable}@<T> && std::@\stdconcept{equality_comparable}@<T>;
+concept @\defexposconceptnc{Vector}@ = is_vector<T>;    // \expos
 
 template<typename T>
-concept @\deflibconcept{Representation}@ = (is_scalar<T> || is_complex<T> || is_vector<T> || is_tensor<T>) &&
-                         @\exposconceptnc{WeaklyRegular}@<T> && @\exposconceptnc{Scalable}@<T>;
+concept @\defexposconceptnc{Tensor}@ = is_tensor<T>;    // \expos
 
 template<typename T, quantity_character Ch>
-concept @\deflibconcept{RepresentationOf}@ =
-  @\libconcept{Representation}@<T> && ((Ch == quantity_character::scalar && is_scalar<T>) ||
+concept @\defexposconceptnc{IsOfCharacter}@ = (Ch == quantity_character::scalar && is_scalar<T>) ||  // \expos
                         (Ch == quantity_character::complex && is_complex<T>) ||
                         (Ch == quantity_character::vector && is_vector<T>) ||
-                        (Ch == quantity_character::tensor && is_tensor<T>));
+                        (Ch == quantity_character::tensor && is_tensor<T>);
+
+template<typename T>
+using @\exposidnc{scaling-factor-type-t}@ =  // \expos
+  conditional<treat_as_floating_point<T>, long double, std::intmax_t>;
+\end{itemdecl}
+
+\begin{itemdecl}
+template<typename T>
+concept @\defexposconceptnc{ScalarRepresentation}@ =  // \expos
+  @\exposconceptnc{Scalar}@<T> && @\exposconceptnc{WeaklyRegular}@<T> && requires(T a, T b, @\exposidnc{scaling-factor-type-t}@<T> f) {
+    // scaling
+    { a * f } -> Scalar;
+    { f * a } -> Scalar;
+    { a / f } -> Scalar;
+
+    // scalar operations
+    { -a } -> Scalar;
+    { a + b } -> Scalar;
+    { a - b } -> Scalar;
+    { a * b } -> Scalar;
+    { a / b } -> Scalar;
+  };
+\end{itemdecl}
+
+\pnum
+A type \tcode{T} models \libconcept{ScalarRepresentation} if
+TBD.
+
+\begin{itemdecl}
+template<typename T>
+concept @\defexposconceptnc{ComplexRepresentation}@ =  // \expos
+  @\exposconceptnc{Complex}@<T> && @\exposconceptnc{WeaklyRegular}@<T> && requires(T a, T b, @\exposidnc{scaling-factor-type-t}@<T> f) {
+    // scaling
+    { a * T(f) } -> Complex;
+    { T(f) * a } -> Complex;
+    { a / T(f) } -> Complex;
+
+    // complex operations
+    { -a } -> Complex;
+    { a + b } -> Complex;
+    { a - b } -> Complex;
+    { a * b } -> Complex;
+    { a / b } -> Complex;
+    { real(a) } -> Scalar;
+    { imag(a) } -> Scalar;
+    { abs(a) } -> Scalar;
+    { arg(a) } -> Scalar;
+    { conj(a) } -> Complex;
+  };
 \end{itemdecl}
 
+\pnum
+A type \tcode{T} models \libconcept{ComplexRepresentation} if
+TBD.
+
+\begin{itemdecl}
+template<typename T>
+concept @\defexposconceptnc{VectorRepresentation}@ =  // \expos
+  @\exposconceptnc{Vector}@<T> && @\exposconceptnc{WeaklyRegular}@<T> && requires(T a, T b, @\exposidnc{scaling-factor-type-t}@<T> f) {
+    // scaling
+    { a * f } -> Vector;
+    { f * a } -> Vector;
+    { a / f } -> Vector;
```

Comment:
`scaling-factor-type-t` and scaling checks will be replaced by the work done by Yves. Did you look into it? There are a few things that potentially could be improved. It would be great if you could review his PR before we merge it.

Reply:
I had no idea.